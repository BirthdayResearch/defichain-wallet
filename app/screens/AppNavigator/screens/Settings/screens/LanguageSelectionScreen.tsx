import { SectionTitle } from '@components/SectionTitle'
import { ThemedScrollView } from '@components/themed'
import { getAppLanguages, translate } from '@translations'
import * as React from 'react'
import { RowLanguageItem } from '../components/RowLanguageItem'

export function LanguageSelectionScreen (): JSX.Element {
  const languages = getAppLanguages()

  return (
    <ThemedScrollView>
      <SectionTitle
        testID='language_selection_screen_title'
        text={translate('screens/LanguageSelectionScreen', 'LANGUAGE')}
      />

      {
        languages.map((language, index) => (
          <RowLanguageItem
            key={index}
            language={language}
          />
        ))
      }
    </ThemedScrollView>
  )
}
