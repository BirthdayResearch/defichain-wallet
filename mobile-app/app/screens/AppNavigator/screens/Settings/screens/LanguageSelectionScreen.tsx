import { ThemedSectionTitle } from '@components/themed/ThemedSectionTitle'
import { getAppLanguages, translate } from '@translations'
import { View } from 'react-native'
import { RowLanguageItem } from '../components/RowLanguageItem'

export function LanguageSelectionScreen (): JSX.Element {
  const languages = getAppLanguages()

  return (
    <View testID='language_selection_screen'>
      <ThemedSectionTitle
        testID='language_selection_screen_title'
        text={translate('screens/LanguageSelectionScreen', 'LANGUAGE')}
      />

      {
        languages.map((language, index) => (
          <RowLanguageItem
            key={index}
            languageItem={language}
          />
        ))
      }
    </View>
  )
}
