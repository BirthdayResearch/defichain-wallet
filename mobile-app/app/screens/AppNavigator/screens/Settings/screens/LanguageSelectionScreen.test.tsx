import { render } from '@testing-library/react-native'
import { LanguageSelectionScreen } from './LanguageSelectionScreen'

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn()
}))

jest.mock('@shared-contexts/LanguageProvider')
jest.mock('@shared-contexts/ThemeProvider')

describe('language selection screen', () => {
  it('should render', async () => {
    const rendered = render(<LanguageSelectionScreen />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
