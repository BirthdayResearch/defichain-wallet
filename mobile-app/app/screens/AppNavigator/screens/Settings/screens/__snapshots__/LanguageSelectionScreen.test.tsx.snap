// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`language selection screen should render 1`] = `
<View
  testID="language_selection_screen"
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Array [
          Object {
            "color": "rgba(115, 115, 115, 1)",
            "fontFamily": "MediumFont",
            "fontSize": 12,
            "fontWeight": "500",
            "lineHeight": 16,
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 24,
          },
          Object {
            "backgroundColor": "rgba(245, 245, 245, 1)",
            "color": "rgba(115, 115, 115, 1)",
          },
        ],
      ]
    }
    testID="language_selection_screen_title"
    text="LANGUAGE"
  >
    LANGUAGE
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(229, 229, 229, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(229, 229, 229, 1)",
        "borderRightColor": "rgba(229, 229, 229, 1)",
        "borderTopColor": "rgba(229, 229, 229, 1)",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 8,
        "paddingTop": 16,
      }
    }
    testID="button_language_English"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontFamily": "MediumFont",
                "fontWeight": "500",
              },
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
        testID="language_option"
      >
        English
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontSize": 14,
                "lineHeight": 20,
              },
              Object {
                "color": "rgba(115, 115, 115, 1)",
              },
            ],
          ]
        }
        testID="language_option_description"
      >
        English
      </Text>
    </View>
    <Text />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(229, 229, 229, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(229, 229, 229, 1)",
        "borderRightColor": "rgba(229, 229, 229, 1)",
        "borderTopColor": "rgba(229, 229, 229, 1)",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 8,
        "paddingTop": 16,
      }
    }
    testID="button_language_German"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontFamily": "MediumFont",
                "fontWeight": "500",
              },
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
        testID="language_option"
      >
        Deutsch
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontSize": 14,
                "lineHeight": 20,
              },
              Object {
                "color": "rgba(115, 115, 115, 1)",
              },
            ],
          ]
        }
        testID="language_option_description"
      >
        German
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(229, 229, 229, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(229, 229, 229, 1)",
        "borderRightColor": "rgba(229, 229, 229, 1)",
        "borderTopColor": "rgba(229, 229, 229, 1)",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 8,
        "paddingTop": 16,
      }
    }
    testID="button_language_Chinese (Simplified)"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontFamily": "MediumFont",
                "fontWeight": "500",
              },
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
        testID="language_option"
      >
        简体中文
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontSize": 14,
                "lineHeight": 20,
              },
              Object {
                "color": "rgba(115, 115, 115, 1)",
              },
            ],
          ]
        }
        testID="language_option_description"
      >
        Chinese (Simplified)
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(229, 229, 229, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(229, 229, 229, 1)",
        "borderRightColor": "rgba(229, 229, 229, 1)",
        "borderTopColor": "rgba(229, 229, 229, 1)",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 8,
        "paddingTop": 16,
      }
    }
    testID="button_language_Chinese (Traditional)"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontFamily": "MediumFont",
                "fontWeight": "500",
              },
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
        testID="language_option"
      >
        繁體中文
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontSize": 14,
                "lineHeight": 20,
              },
              Object {
                "color": "rgba(115, 115, 115, 1)",
              },
            ],
          ]
        }
        testID="language_option_description"
      >
        Chinese (Traditional)
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderBottomColor": "rgba(229, 229, 229, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "rgba(229, 229, 229, 1)",
        "borderRightColor": "rgba(229, 229, 229, 1)",
        "borderTopColor": "rgba(229, 229, 229, 1)",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 8,
        "paddingTop": 16,
      }
    }
    testID="button_language_French"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontFamily": "MediumFont",
                "fontWeight": "500",
              },
              Object {
                "color": "rgba(0, 0, 0, 1)",
              },
            ],
          ]
        }
        testID="language_option"
      >
        Français
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "fontSize": 14,
                "lineHeight": 20,
              },
              Object {
                "color": "rgba(115, 115, 115, 1)",
              },
            ],
          ]
        }
        testID="language_option_description"
      >
        French
      </Text>
    </View>
  </View>
</View>
`;
